main.rb

```diff
@@ -0,0 +1,172 @@
+module Enumerable
+  def my_each
+    pos = 0
+    loop do
+      break if pos >= length
+
+      yield self[pos]
+      pos += 1
+    end
+  end
+
+  def my_each_with_index
+    pos = 0
+    loop do
+      break if pos >= length
+
+      yield self[pos], pos
+      pos += 1
+    end
+  end
+
+  def my_select
+    return_arr = []
+    my_each do |element|
+      return_arr.push(element) if yield element
+    end
+    return_arr
+  end
+
+  def my_all
+    return_val = true
+    my_each do |element|
+      return_val = yield element
+    end
+    return_val
+  end
+
+  def my_any
+    return_val = false
+    my_each do |element|
+      return_val = yield element
+    end
+    return_val
+  end
+
+  def my_none
+    return_val = true
+    my_each do |element|
+      return_val = !(yield element)
+    end
+    return_val
+  end
```

Comment:
The issues with **my_none?**, **my_any?** and **my_all?** are similar. the methods should be able to: 

* [ ] Handle the cases where no block is given (Hint: block_given?)
* [ ] Accept primitives, classes (i.e Integer, String, etc...) and Regex Expressions as arguments.